Tests/SonarScanner.MSBuild.Shim.Test/PathHelperTests.cs

```diff
@@ -105,84 +106,126 @@ public void IsInDirectory_ReturnsTheExpectedResults()
+                    new DirectoryInfo(@"D:\ThreeTimes\B"),
+                    new DirectoryInfo(@"D:\ThreeTimes\C"),
+                    new DirectoryInfo(@"E:\Two\A"),
+                    new DirectoryInfo(@"E:\Two\B"),
+                },
+                StringComparer.Ordinal).FullName.Should().Be(@"D:\ThreeTimes");
 
         [TestMethod]
         public void BestCommonPrefix_WhenCommonPath_ReturnsTheLongestCommonPart() =>
-            PathHelper.BestCommonPrefix(new[]
-            {
-                new DirectoryInfo(@"C:\Common"),
-                new DirectoryInfo(@"C:\Common\SubDirA"),
-                new DirectoryInfo(@"C:\Common\SomethingElse"),
-            }).FullName.Should().Be(@"C:\Common");
+            PathHelper.BestCommonPrefix(
+                new[]
+                {
+                    new DirectoryInfo(@"C:\Common"),
+                    new DirectoryInfo(@"C:\Common\SubDirA"),
+                    new DirectoryInfo(@"C:\Common\SomethingElse"),
+                },
+                StringComparer.Ordinal).FullName.Should().Be(@"C:\Common");
 
         [TestMethod]
         public void BestCommonPrefix_WhenCommonPathOfFiles_ReturnsTheLongestCommonPart() =>
-            PathHelper.BestCommonPrefix(new[]
+            PathHelper.BestCommonPrefix(
+                new[]
+                {
+                    new DirectoryInfo(@"C:\InRoot.cs"),
+                    new DirectoryInfo(@"C:\SubDir\A.cs"),
+                    new DirectoryInfo(@"C:\SubDir\B.cs"),
+                },
+                StringComparer.Ordinal).FullName.Should().Be(@"C:\");
+
+        [TestMethod]
+        [DynamicData(nameof(CommonPrefixCasing))]
+        public void BestCommonPrefix_IgnoreCasingComparer(string[] paths, StringComparer comparer, string expected)
+        {
+            var commonPrefix = PathHelper.BestCommonPrefix(paths.Select(x => new DirectoryInfo(x)), comparer);
+            if (expected is null)
+            {
+                commonPrefix.Should().BeNull();
+            }
+            else
+            {
+                commonPrefix.FullName.Should().Be(expected);
+            }
+        }
+
+        public static IEnumerable<object[]> CommonPrefixCasing =>
+            new[]
             {
-                new DirectoryInfo(@"C:\InRoot.cs"),
-                new DirectoryInfo(@"C:\SubDir\A.cs"),
-                new DirectoryInfo(@"C:\SubDir\B.cs"),
-            }).FullName.Should().Be(@"C:\");
+                new object[] { new[] { @"c:\InRoot.cs", @"C:\SubDir\A.cs" }, StringComparer.OrdinalIgnoreCase, @"c:\" },
+                new object[] { new[] { @"c:\InRoot.cs", @"C:\SubDir\A.cs" }, StringComparer.Ordinal, null },
+                new object[] { new[] { @"c:\InRoot.cs", @"C:\SubDir\A.cs" }, StringComparer.InvariantCultureIgnoreCase, @"c:\" },
+                new object[] { new[] { @"c:\InRoot.cs", @"C:\SubDir\A.cs" }, StringComparer.InvariantCulture, null }
```

Comment:
Same comment as above, but here I would insist a bit more since it is way shorter and more readable.
You can eliminate both the outer-`new[]` and the inner `new object[]`:

```csharp
   public static IEnumerable<object[]> CommonPrefixCasing =>
    [
        [new[] { @"c:\InRoot.cs", @"C:\SubDir\A.cs" }, StringComparer.OrdinalIgnoreCase, @"c:\"],
        [new[] { @"c:\InRoot.cs", @"C:\SubDir\A.cs" }, StringComparer.Ordinal, null],
        [new[] { @"c:\InRoot.cs", @"C:\SubDir\A.cs" }, StringComparer.InvariantCultureIgnoreCase, @"c:\"],
        [new[] { @"c:\InRoot.cs", @"C:\SubDir\A.cs" }, StringComparer.InvariantCulture, null]
    ];
```